### DISCUSSION TITLE
`endOfDay` getter

### DISCUSSION THREAD

**Discussion by FMorschel** [jogboms/time.dart#69] [Ideas]
Just as there are both `date` and `timeOfDay` getters I'm here to suggest a new getter called `endOfDay`. I'm not sure it is going to be a truly used feature though, but for my new `Period` class in due_date, it would certainly be a great help since I'm constantly creating full-day periods. I'm suggesting you implement it in this package only because I believe this feature is more coherent if placed inside your package than mine.

As you can see here, I would be willing to create a PR for this if you like this idea.

```dart
DateTime get endOfDay {
  late DateTime date;
  if (isUtc) {
    date = DateTime.utc(year, month, day + 1);
  } else {
    date = DateTime(year, month, day + 1);
  }
  while (!isAtSameDayAs(date)) {
    date = date.subtract(const Duration(microseconds: 1));
  }
  return date;
}
```

**Comment by jogboms** [comment]
Lets do this! @FMorschel

**Reply by FMorschel** [reply]
Do you think I should try and create tests using [timezone](https://pub.dev/packages/timezone) so we can test daylight savings (the reason why I added that while on the snippet, even tho in retrospect I think it's useless because at that point any subtraction should always go to the other day)? I'm unfamiliar with the package, but I could try to make it work for the tests.

**Reply by FMorschel** [reply]
I guess another question would be what is your opinion on using the while there?

**Reply by jogboms** [reply]
I’m fine with that being a dev dependency and looking forward to seeing the tests to get full grasp of the edge case the loop is solving

**Reply by FMorschel** [reply]
Hello again! After a long time I finally got around working on this again.

I live in Brazil and until recently we had daylight savings that started exactly midnight. So in that case, we had one our less/more to that day. This was what I was trying to grasp with it (the case where we had one our less - that day would end at 23h). I know some other countries have daylight savings starting during the day so those cases would not interfere here.

I tried using the timezone package to see if it covered that case but I was not able to do it. Whether it was looking for the machine to see when that would happened (no daylight savings anymore, our previous government removed them), or if it isn't capable of managing that I'm not aware, but in any case, this was not possible for now.

I will post a PR with the changes without the loop.
